From 1764250fc2147ae69d739c26a1e50d502c635ed2 Mon Sep 17 00:00:00 2001
From: Cameron Rozean <rcrozean@amazon.com>
Date: Thu, 9 Mar 2023 14:24:07 -0800
Subject: [PATCH] Add SSL redirect and default service backend to Athens
 ingress template

The upstream template does not have the SSL redirect rule and the default
service backend configured, so this patch adds it. We need this for the
redirect ingress annotations to work.
Signed-off-by: Abhay Krishna Arunachalam <arnchlm@amazon.com>
---
 charts/athens-proxy/templates/ingress.yaml | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/charts/athens-proxy/templates/ingress.yaml b/charts/athens-proxy/templates/ingress.yaml
index eda575a8..b437b6fc 100644
--- a/charts/athens-proxy/templates/ingress.yaml
+++ b/charts/athens-proxy/templates/ingress.yaml
@@ -26,6 +26,11 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  defaultBackend:
+    service:
+      name: {{ $fullName }}
+      port:
+        number: {{ $svcPort }}
   {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
   ingressClassName: {{ .Values.ingress.className }}
   {{- end }}
@@ -40,10 +45,17 @@ spec:
     {{- end }}
   {{- end }}
   rules:
-    {{- range .Values.ingress.hosts }}
+    {{- range $host := .Values.ingress.hosts }}
     - host: {{ .host | quote }}
       http:
         paths:
+          - path: /
+            pathType: ImplementationSpecific
+            backend:
+              service:
+                name: ssl-redirect
+                port:
+                  number: use-annotation
           {{- range .paths }}
           - path: {{ .path }}
             {{- if and .pathType (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion) }}
-- 
2.39.2

